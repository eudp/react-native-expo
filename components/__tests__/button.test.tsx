import React from "react";
import renderer from "react-test-renderer";
import Button from "../Button";
import { TouchableOpacity } from "react-native";

describe("Button", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Button text="Press me!" onPress={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onPress when pressed", () => {
    const onPress = jest.fn();
    const component = renderer.create(
      <Button text="Press me!" onPress={onPress} />
    );

    const button = component.root.findByType(TouchableOpacity);
    button.props.onPress();
    expect(onPress).toHaveBeenCalled();
  });
});
