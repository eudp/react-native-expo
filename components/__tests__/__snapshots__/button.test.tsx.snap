// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#334FFA",
      "borderRadius": 10,
      "flex": 1,
      "opacity": 1,
      "padding": 10,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Avenir-Regular",
        },
        {
          "color": "#FFFFFF",
          "fontFamily": "Avenir-Black",
        },
      ]
    }
  >
    Press me!
  </Text>
</View>
`;
